mbrl/util/replay_buffer.py

```diff
@@ -151,24 +164,128 @@ def __next__(self):
+    The start states of all trajectories are sampled uniformly at random from the set of
+    states from which a sequence of the desired length can be started.
+
+    When iterating over this object, batches might contain overlapping trajectories. By default,
+    a full loop over this iterator will return as many samples as valid start states
+    there are (but start states could be repeated, they are sampled with replacement). Since
+    this is unlikely necessary, you can use input argument ``batches_per_epoch`` to
+    only return a smaller number of batches.
+
+    Note that this is a bootstrap iterator, so it can return an extra model dimension,
+    where each batch is sampled independently. By default, each observation batch is of
+    shape (ensemble_size x batch_size x sequence_length x obs_size)  -- likewise for
+    actions, rewards, dones. If not in bootstrap mode, then the ensemble_size dimension
+    is removed.
+
+
+    Args:
+        transitions (:class:`TransitionBatch`): the transition data used to built
+            the iterator.
+        trajectory_indices (list(tuple(int, int)): a list of [start, end) indices for
+            trajectories.
+        batch_size (int): the batch size to use when iterating over the stored data.
+        sequence_length (int): the length of the sequences returned.
+        ensemble_size (int): the number of models in the ensemble.
+        rng (np.random.Generator, optional): a random number generator when sampling
+            batches. If ``None`` (default value), a new default generator will be used.
+        max_batches_per_loop (int, optional): if given, specifies how many batches
+            to return (at most) over a full loop of the iterator.
+    """
+
+    def __init__(
+        self,
+        transitions: TransitionBatch,
+        trajectory_indices: List[Tuple[int, int]],
+        batch_size: int,
+        sequence_length: int,
+        ensemble_size: int,
+        rng: Optional[np.random.Generator] = None,
+        max_batches_per_loop: Optional[int] = None,
+    ):
+        self._sequence_length = sequence_length
+        self._valid_starts = self._get_indices_valid_starts(
+            trajectory_indices, sequence_length
+        )
+        self._max_batches_per_loop = max_batches_per_loop
+        if len(self._valid_starts) < 0.5 * len(trajectory_indices):
+            warnings.warn(
+                "More than 50% of the trajectories were discarded for being shorter "
+                "than the specified length."
+            )
+        # no need to pass transitions to super(), since it's only used by __getitem__,
+        # which this class replaces. Passing the set of possible starts allow us to
+        # use all the indexing machinery of the superclasses.
+        super().__init__(
+            self._valid_starts,  # type: ignore
+            batch_size,
+            ensemble_size,
+            shuffle_each_epoch=False,
```

Comment:
Is there a reason why ``shuffle_each_epoch=False`` is hardcoded here? I would prefer being able to pass shuffle_each_epoch as an argument to the constructor of SequenceTransitionIterator.
I tried this out by setting ``_shuffle_each_epoch = True`` from outside and it seemed to work fine.

Reply:
Ah, good catch! At some point while I was working on this I had a version where this would have messed up the trajectories, but this is no longer relevant as it only shuffles the order of start states. I'll add back this option.

Reply:
Perfect, thanks

Reply:
Added in https://github.com/facebookresearch/mbrl-lib/pull/91/commits/029e0ab84bb9bf9f41da980305dde085b3bb74ff